server_addon/maya/server/settings/publishers.py

```diff
@@ -309,6 +385,143 @@ class ExtractAlembicModel(BaseSettingsModel):
         description="List of attribute prefixes for attributes that will be "
                     "included in the alembic export.",
     )
+    flags: list[str] = SettingsField(
+        enum_resolver=extract_alembic_flags_enum,
+        title="Export Flags",
+        description=(
+            "Auto Subd - If this flag is present and the mesh has crease "
+            "edges, crease vertices or holes, the mesh (OPolyMesh) would now "
+            "be written out as an OSubD and crease info will be stored in the "
+            "Alembic  file. Otherwise, creases info won't be preserved in "
+            "Alembic file unless a custom Boolean attribute SubDivisionMesh "
+            "has been added to mesh node and its value is true.\n"
+
+            "Dont Skip Unwritten Frames - When evaluating multiple translate "
+            "jobs, this decides whether to evaluate frames between jobs when "
+            "there is a gap in their frame ranges.\n"
+
+            "Euler Filter - Apply Euler filter while sampling rotations.\n"
+
+            "No Normals - Present normal data for Alembic poly meshes will not"
+            " be written.\n"
+
+            "Pre Roll - This frame range will not be sampled.\n"
+
+            "Renderable Only - Only export renderable visible shapes.\n"
+
+            "Strip Namespaces - Namespaces will be stripped off of the node "
+            "before being written to Alembic.\n"
+
+            "UV Write - Uv data for PolyMesh and SubD shapes will be written "
+            "to the Alembic file.\n"
+
+            "UVs Only - If this flag is present, only uv data for PolyMesh and"
+            " SubD shapes will be written to the Alembic file.\n"
+
+            "Verbose - Prints the current frame that is being evaluated.\n"
+
+            "Whole Frame Geo - Data for geometry will only be written out on "
+            "whole frames.\n"
+
+            "World Space - Any root nodes will be stored in world space.\n"
+
+            "Write Color Sets - Write vertex colors with the geometry.\n"
+
+            "Write Face Sets - Write face sets with the geometry.\n"
+
+            "Write Normals - Write normals with the deforming geometry.\n"
+
+            "Write UV Sets - Write all uv sets on MFnMeshes as vector 2 "
+            "indexed geometry parameters with face varying scope.\n"
+
+            "Write Visibility - Visibility state will be stored in the Alembic"
+            " file. Otherwise everything written out is treated as visible."
+        )
```

Comment:
Can we make the labels bold by using `**` around them (markdown bold)
```suggestion
        description=(
            "**Auto Subd** - If this flag is present and the mesh has crease "
            "edges, crease vertices or holes, the mesh (OPolyMesh) would now "
            "be written out as an OSubD and crease info will be stored in the "
            "Alembic  file. Otherwise, creases info won't be preserved in "
            "Alembic file unless a custom Boolean attribute SubDivisionMesh "
            "has been added to mesh node and its value is true.\n"

            "**Dont Skip Unwritten Frames** - When evaluating multiple "
            "translate jobs, this decides whether to evaluate frames between "
            "jobs when there is a gap in their frame ranges.\n"

            "**Euler Filter** - Apply Euler filter while sampling rotations.\n"

            "**No Normals** - Present normal data for Alembic poly meshes will"
            " not be written.\n"

            "**Pre Roll** - This frame range will not be sampled.\n"

            "**Renderable Only** - Only export renderable visible shapes.\n"

            "**Strip Namespaces** - Namespaces will be stripped off of the "
            "node before being written to Alembic.\n"

            "**UV Write** - Uv data for PolyMesh and SubD shapes will be "
            "written to the Alembic file.\n"

            "**UVs Only** - If this flag is present, only uv data for PolyMesh"
            " and SubD shapes will be written to the Alembic file.\n"

            "**Verbose** - Prints the current frame that is being evaluated.\n"

            "**Whole Frame Geo** - Data for geometry will only be written out "
            "on whole frames.\n"

            "**World Space** - Any root nodes will be stored in world space.\n"

            "**Write Color Sets** - Write vertex colors with the geometry.\n"

            "**Write Face Sets** - Write face sets with the geometry.\n"

            "**Write Normals** - Write normals with the deforming geometry.\n"

            "**Write UV Sets** - Write all uv sets on MFnMeshes as vector 2 "
            "indexed geometry parameters with face varying scope.\n"

            "**Write Visibility** - Visibility state will be stored in the "
            "Alembic file. Otherwise everything written out is treated as "
            "visible."
        )
```

Reply:
Also, I'm pretty sure the description for **Pre Roll** is incorrect? _This frame range will not be sampled_ seems pretty wrong?

Reply:
> Also, I'm pretty sure the description for Pre Roll is incorrect? This frame range will not be sampled seems pretty wrong?

Think this was from the alembic export help descriptions. Think they mean that the pre roll range will not become part of the export, so not written to disk.

Reply:
Interesting - since the flag I suppose would be the reverse, when enabled then it _does_ get written? Anyway, let's rewrite it to be clearer: When enabled, the pre roll start frame is used to pre roll the export for.

Side question - is the "pre roll" just sampled for the export so that it runs from there. Or is it actually included and written into the alembic as well?

Reply:
> Side question - is the "pre roll" just sampled for the export so that it runs from there. Or is it actually included and written into the alembic as well?

Think pre roll is just sampled for the export. Think its meant for simulation run up.

Reply:
> Interesting - since the flag I suppose would be the reverse, when enabled then it does get written? Anyway, let's rewrite it to be clearer: When enabled, the pre roll start frame is used to pre roll the export for.

Done